core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -527,10 +527,11 @@ public static boolean updateTableStatusFile(CarbonTable carbonTable, String segm
         for (LoadMetadataDetails detail : listOfLoadFolderDetailsArray) {
           // if the segments is in the list of marked for delete then update the status.
           if (segmentId.equals(detail.getLoadName())) {
+            detail.setLoadEndTime(System.currentTimeMillis());
```

Comment:
why we need this change? is it sepicific to non carbon segment?

Reply:
yes

Reply:
is this common code to both carbon and non carbon segments? if it is  then better to set it only in non carbon segments case.

Reply:
I think we should not add external segments check for common things, it should be okay if we set here because we are not using this end time in any other logic/decision making